packages/react-html/src/components/Link.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import {Effect} from '@shopify/react-effect';
+import {Consumer} from '../context';
+import {EFFECT_ID} from '../utilities';
+import Manager from '../manager';
+
+type Props = React.HTMLProps<HTMLLinkElement>;
+
+export default class Link extends React.PureComponent<Props> {
+  private removeLink?: ReturnType<InstanceType<typeof Manager>['addLink']>;
+
+  componentWillUnmount() {
+    if (this.removeLink) {
+      this.removeLink();
+    }
+  }
+
+  render() {
+    return (
+      <Consumer>
+        {manager => (
+          <Effect
+            key={JSON.stringify(this.props)}
+            kind={EFFECT_ID}
+            perform={() => {
+              if (this.removeLink) {
+                this.removeLink();
```

Comment:
I get that this imperative manager api mostly exists to support the declarative api, but I wonder if it might be nicer to have the manager api also declarative? 

```
<Consumer>
  {(manager) => {
    const {Head} = manager;
    return (
      <Head>
        <link />
      </Head>
    );
 }
</Consumer>
```

This way when / if consumers try to use the manager directly for this sort of thing (I'm sure there will be cases where this has to happen for some reason or another) they still get a nice imperative api without having to use `Effect` or manage callbacks.

Reply:
Talked IRL, since `Consumer` is not really exposed publicly I think it is OK to keep the API as-is.